lib/turn.rb

```diff
@@ -0,0 +1,88 @@
+class Turn 
+attr_accessor :spoils_of_war, 
+              :type, 
+              :pile_cards,
+              :award_spoils, 
+              :player1, 
+              :player2, 
+              :winner
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+ 
+  def type 
+    if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+      :basic
+    elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
+      if @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
+        :war
+      elsif @player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)
+        :mutually_assured_destruction
+      elsif @player1.deck.rank_of_card_at(2) != @player2.deck.rank_of_card_at(2)
+        :war
+      end
+    end
```

Comment:
Here's a human-readable summary of the conditions we want to check for:
- :basic means player 1 and player 2 do NOT have the same rank_of_card_at(0) values
- :war means both players’ rank_of_card_at(0) are the same
- :mutually_assured_destruction means both players' rank_of_card_at(0) AND both players' rank_of_card_at(2) are the same

Imagine a world where we have lots of helper methods to break this down:
```
# this method returns 'true' when both players’ rank_of_card_at(0) AND rank_of_card_at(2) are the same.
def mutually_assured_destruction? 
  (@player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)) &&
    (@player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2))
end
```
```
# this method returns 'true' when both players’ rank_of_card_at(0) are the same
def war?
  @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
end
```
```
if mutually_assured_destruction? # war and basic conditions would both evaluate to false here
  :mutually_assured_destruction
elsif war?
  :war
else # if it's not mutually assured destruction, and it's not war, it must be basic
  :basic
end
```